lib/AWS/Signature4.pm

```diff
@@ -240,6 +240,9 @@ sub _scope {
     } elsif ($host =~ /^(\w+)[-.]([\w-]+)\.amazonaws\.com/) {
 	$service  = $1;
 	$region ||= $2;
+    } elsif ($host =~ /^([\w-]+)\.([\w-]+)\.amazonaws\.com/) {
+	$service  = $1;
+	$region ||= $2;
```

Comment:
I came upon this PR after hitting the same bug with ec2-instance-connect endpoints.
https://docs.aws.amazon.com/general/latest/gr/ec2-instance-connect.html

Your new regex looks correct to me. Should it not _replace_ the regex in the conditional above, though? The intent seems to be to match regional endpoints as documented here:
https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints

I'm new to this, so I'm not sure, but I think the original regex is just incorrect.

Reply:
:+1:  for this, just ran into this problem with bedrock-runtime. Testing with replacing lines 240-242 with the adjusted regex works for me, so I tend to agree with @bugfood .